WordPress/Classes/ViewRelated/Reader/Tab Navigation/WPTabBarController+ReaderTabNavigation.swift

```diff
@@ -1,9 +1,107 @@
-import UIKit
+/// Generic type for the UIViewController in the Reader Content View
+protocol ReaderContentViewController: UIViewController {
+    func setTopic(_ topic: ReaderAbstractTopic?)
+}
+
 
+/// Reader Factory
 extension WPTabBarController {
 
+    var readerTabViewController: ReaderTabViewController? {
+        readerNavigationController?.topViewController as? ReaderTabViewController
+    }
+
     @objc func makeReaderTabViewController() -> ReaderTabViewController {
-        let viewModel = ReaderTabViewModel()
-        return ReaderTabViewController(viewModel: viewModel)
+        return ReaderTabViewController(viewModel: self.readerTabViewModel, readerTabViewFactory: makeReaderTabView(_:))
+    }
+
+    @objc func makeReaderTabViewModel() -> ReaderTabViewModel {
+        let viewModel = ReaderTabViewModel(readerContentFactory: makeReaderContentViewController(with:),
+                                           searchNavigationFactory: navigateToReaderSearch)
+        return viewModel
+    }
+
+    func makeReaderContentViewController(with topic: ReaderAbstractTopic?) -> ReaderContentViewController {
+        var controller: ReaderStreamViewController
+        if let topic = topic {
+            controller = ReaderStreamViewController.controllerWithTopic(topic)
+        } else {
+            controller = ReaderStreamViewController.controllerForSavedPosts()
+        }
+        return controller
+    }
+
+    private func makeReaderTabView(_ viewModel: ReaderTabViewModel) -> ReaderTabView {
+        return ReaderTabView(viewModel: self.readerTabViewModel)
+    }
+}
+
+
+/// Reader Navigation
+extension WPTabBarController {
+
+    /// reader navigation methods
```

Comment:
These comments might be better as `// MARK: - Navigation` style comments, to group the various methods together.